### PR TITLE
[SPARK-42078][PYTHON][FOLLOWUP] Add `CapturedException` to utils

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This is follow-up for https://github.com/apache/spark/pull/39591 based on the suggestions from comment https://github.com/apache/spark/commit/1de83500f4621d4af91f25a339dd5057c59cfc1e#r109023188.

### Why are the changes needed?

For backward compatibility.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

The existing CI should pass

### CODE REVIEW

**Comment by itholic** [comment]
cc @mingyangge-db @HyukjinKwon FYI

File: python/pyspark/sql/utils.py [hunk 1]
```diff
@@ -37,6 +37,7 @@
     PythonException,
     UnknownException,
     SparkUpgradeException,
+    CapturedException,
```
Comment (mingyangge-db) [inline python/pyspark/sql/utils.py:40]: Now that 3.4.0 is already published. This change will only be available in next version, right?

File: python/pyspark/errors/__init__.py [hunk 2]
```diff
@@ -59,4 +61,5 @@
     "PySparkTypeError",
     "PySparkValueError",
     "PySparkAttributeError",
+    "CapturedException",
```
Comment (HyukjinKwon) [inline python/pyspark/errors/__init__.py:64]: Can you remove this here but add it into `python/pyspark/sql/utils.py` alone? Adding `__all__` implies that it's an API

File: python/pyspark/sql/utils.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/utils.py:40]: Yes.

Comment (HyukjinKwon) [reply python/pyspark/sql/utils.py:40]: For now, you would work around via try-catch with import.

File: python/pyspark/errors/__init__.py (hunk 2 above)
Comment (itholic) [reply python/pyspark/errors/__init__.py]: Updated

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.4.

**Pull request closed without merging**